pac4j-core/src/main/java/org/pac4j/core/credentials/extractor/HeaderExtractor.java

```diff
@@ -22,17 +24,17 @@ public HeaderExtractor(final String headerName, final String prefixHeader) {
     }
 
     @Override
-    public TokenCredentials extract(WebContext context) {
+    public Optional<TokenCredentials> extract(WebContext context) {
         final String header = context.getRequestHeader(this.headerName);
```

Comment:
I'm surprised that getRequestHeader does not return an optional :)

Reply:
Same as above...

Reply:
same as above :)